### PR TITLE
Add zsh completion for 'dockerd --init' and 'docker {create,run} --init'

### PR DESCRIPTION
**Pull request by sdurrheimer**
ref. #26061

Signed-off-by: Steve Durrheimer <s.durrheimer@gmail.com>

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: contrib/completion/zsh/_docker [hunk 1]
```diff
@@ -541,6 +541,7 @@ __docker_container_subcommand() {
         "($help)*--group=[Set one or more supplementary user groups for the container]:group:_groups"
         "($help -h --hostname)"{-h=,--hostname=}"[Container host name]:hostname:_hosts"
         "($help -i --interactive)"{-i,--interactive}"[Keep stdin open even if not attached]"
+        "($help)--init[Run an init inside the container that forwards signals and reaps processes]" \
```
Comment (thaJeztah) [inline contrib/completion/zsh/_docker]: Should the trailing backslash be here?

Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: You're right. Will fix this soonish.

Comment (vieux) [inline contrib/completion/zsh/_docker]: @thaJeztah @sdurrheimer is the fix somewhere ?

Comment (thaJeztah) [inline contrib/completion/zsh/_docker]: @vieux think it's in https://github.com/docker/docker/issues/30396

Comment (vieux) [inline contrib/completion/zsh/_docker]: thanks, I'll wait for this one to be merged then.

**Pull request merged**